R/pmcmc_state.R

```diff
@@ -220,11 +229,20 @@ pmcmc_state <- R6::R6Class(
           private$update_history_nested()
         }
 
-        if (i %% 2) {
-          private$run_fixed()
+        if (!is.null(fix_vary_step)) {
```

Comment:
This (combined with the above) is pretty gnarly.

I wonder if you could pull it into a closure so that the interface would look like:

```r
run <- alternate(private$run_fixed, private$run_varied, private$control$nested_step_ratio)
...
run(j)
```

by doing something like (untested!):

```r
alternate <- function(f, g, ratio) {
  if (ratio < 1) {
    return(alternate, g, f, 1 / ratio)
  }
  assert_integer(ratio)
  function(i) {
    if (i %% (ratio + 1) == 0) {
      g()
    } else {
      f()
    }
  }
}
```